libs/SalesforceHybridSDK/SalesforceHybridSDK/Classes/SFHybridViewController.m

```diff
@@ -425,6 +425,15 @@ - (NSURL *)frontDoorUrlWithReturnUrl:(NSString *)returnUrlString returnUrlIsEnco
     NSURL *instUrl = creds.apiUrl;
     NSString *fullReturnUrlString = returnUrlString;
     
+    // Catch if we have a absoute URL, if we do then url decode to avoid
+    // errors with 'Invalid page redirection'
+    if(createAbsUrl && [returnUrlString hasPrefix:@"http"]) {
+        NSString *decodedUrlString = [(NSString *)returnUrlString stringByReplacingOccurrencesOfString:@"+" withString:@" "];
```

Comment:
I don't think you need the explicit cast here - `returnUrlString` is of type `NSString *`.

Reply:
Ah OK, thanks